vector/README.md

```diff
@@ -0,0 +1,110 @@
+```
+$ cat blocks_471000-471010.json | jq ".parameters"
+{
+ "from": 471000,
+ "to": 471010,
+ "tasks": [
+   "blocks"
+ ],
+ "address-filter": ""
+}
+```
+the extracted chain state as a base64 encoded CAR file:
+```
+$ cat blocks_471000-471010.json | jq ".car"
+<lot_of_data>
+```
+and the models extracted:
+```
+$ cat blocks_471000-471010.json | jq ".expected[].block_headers"
+[
+ {
+   "Height": 471010,
+   "Cid": "bafy2bzacecq2fysiktcuc5r762hvuxstwv4qzx645ld4h66spenheckyuko2y",
+   "Miner": "f02770",
+   "ParentWeight": "10230517226",
+   "ParentBaseFee": "3275882399",
+   "ParentStateRoot": "bafy2bzacecsckqgj6ufbefj7bx6odzvvbftky77py5x2th3wrqx3yvpksyuc6",
+   "WinCount": 1,
+   "Timestamp": 1612436700,
+   "ForkSignaling": 0
+ },
+ {
+   "Height": 471010,
+   "Cid": "bafy2bzacecxmra4cdbqq7lxao7e2gmxvf7ef3o464et7ejykcn24itps7vnau",
+   "Miner": "f02775",
+   "ParentWeight": "10230517226",
+   "ParentBaseFee": "3275882399",
+   "ParentStateRoot": "bafy2bzacecsckqgj6ufbefj7bx6odzvvbftky77py5x2th3wrqx3yvpksyuc6",
+   "WinCount": 1,
+   "Timestamp": 1612436700,
+   "ForkSignaling": 0
+ },
+ ...
+```
+
+## How to execute a vector
+To execute a vector, use the `vector execute` command. The example below executes a test vector file named "blocks_471000-471010.json".
+```
+$ ./visor vector execute --vector-file=blocks_471000-471010.json
+2021-02-16T16:32:29.225-0800   INFO   vector vector/runner.go:146 Validate Model block_headers: Passed
+2021-02-16T16:32:29.230-0800   INFO   vector vector/runner.go:146 Validate Model block_parents: Passed
+2021-02-16T16:32:29.230-0800   INFO   vector vector/runner.go:146 Validate Model drand_block_entries: Passed
+```
+When executing the vector file Visor uses the data in the `car` field as its data source, executes the commands in the `parameters` field, then validates the data returned from the execution matches the `expected` models in the vector file.
+
+## How to save a vector
+Vector files are stored on the IPFS network, and a list of their hashes is kept in the `VECTOR_MANIFEST` file. Storing the hashes in the git repo and the files in IPFS helps keep our repo compact. The example below will demonstrate how to save a vector file to run as a part of CI. We will use the aforementioned `blocks_471000-471010.json` in this example.
+```
+$ ipfs add blocks_471000-471010.json
+added QmPTKSr1uwExGTVByDUoVm4CfrjU34oNV6gfvWvufaN9h1 blocks_471000-471010.json
+ 146.43 KiB / 146.43 KiB [==] 100.00%
+```
+next, add the hash to the manifest:
+```
+$ echo QmPTKSr1uwExGTVByDUoVm4CfrjU34oNV6gfvWvufaN9h1 >> ./vector/VECTOR_MANIFEST
+```
+then make the vector deps:
+```
+$ make deps
+cd ./vector; ./fetch_vectors.sh
+Fetching QmPTKSr1uwExGTVByDUoVm4CfrjU34oNV6gfvWvufaN9h1
+ ```
+You can see the vector file is in `vectors/data/`:
+```
+ls vector/data/
+QmPTKSr1uwExGTVByDUoVm4CfrjU34oNV6gfvWvufaN9h1_block_models_from_epoch_471000_to_471010.json
+```
+Finally, execute the vector unit tests:
+```
+$ go test -v ./vector/...
```

Comment:
Tried these commands out and it all worked smoothly. :trophy:

Reply:
haha, spoke too soon. One vector failed to download correctly and the file contains 

```
<html>
<head><title>429 Too Many Requests</title></head>
<body>
<center><h1>429 Too Many Requests</h1></center>
<hr><center>openresty</center>
</body>
</html>
```

Reply:
Damn, ideally the curl command in `fetch_vectors.sh` would us ipfs or ipget but I didn't want to muck with those in CI just yet. I have modified `fetch_vectors.sh` to fail in the event `curl` fails, and I think that should address this... :2nd_place_medal: